packages/trie/test/proof/rightNode.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { utf8ToBytes } from 'ethereum-cryptography/utils'
+import { describe, expect, it } from 'vitest'
+
+import { BranchNode, Trie } from '../../src/index.js'
+import { hasRightBranch, isValueNode, returnRightNode } from '../../src/proof/range.js'
+import { bytesToNibbles, nibblesCompare, nibblestoBytes } from '../../src/util/nibbles.js'
+describe('returns RightNode', async () => {
+  const trie = new Trie()
+  const entries = Array.from({ length: 20 }, (_, i) => {
+    return {
+      key: utf8ToBytes(`key${i % 3}${i % 4}${i % 5}`),
+      value: Uint8Array.from([i, i, i, i]),
+    }
+  })
+  for (const { key, value } of entries) {
+    await trie.put(key, value)
+  }
+
+  const sortedByKey = entries.sort((a, b) =>
+    nibblesCompare(bytesToNibbles(a.key), bytesToNibbles(b.key))
+  )
+  it('passes', () => {
+    expect(true).toBe(true)
+  })
+  for (const { key, value } of sortedByKey) {
+    it(`returns RightNode for key ${key} (${value})`, async () => {
+      const val = await trie.get(key)
+      const node = (await trie.findPath(key)).node
+      expect(val).toEqual(value)
+      expect(isValueNode(node!)).toBe(true)
+    })
+  }
+})
+
+describe('hasRightBranch', async () => {
+  const branchNode = new BranchNode()
+  for (const i of Array.from({ length: 7 }, (_, i) => i)) {
+    branchNode.setBranch(i, Uint8Array.from([i]))
+  }
+  branchNode.setBranch(8, Uint8Array.from([8]))
+  for (const i of Array.from({ length: 16 }, (_, i) => i)) {
```

Comment:
This is very fancy syntax for `for (let i = 0; i < 16; i++)`

Reply:
> true -- i was under the impression that `for (let i...)` had gone out of fashion.  i am ambivalent

I just have missed the memo. I still do that all the time.

Reply:
Yes, please change back. I couldn't even read this new thing. 😂

Reply:
yeah -- i have no idea... :facepalm:

Reply:
I think this looks somewhat similar to how you use for loops in python (i.e. `range()`) however this is not necessary in JS :)

Reply:
points for creativity

Reply:
> points for creativity

like the points you get for hitting a golf ball into a tree...